### PR TITLE
Add api to get exposed container port from Dockerfile for Github/Bitbucket

### PR DESCRIPTION
**Pull request by akashshinde**
This PR adds support to fetch Dockerfile content using Github and Bitbucket rest api.

### Usage

Get Dockerfile content
```
 const content = gitservice.getDockerfileContent();
```
Get Exposed container port
```
 // Get dockerfile parser
 const parser = new DockerFileParser(content);
 // Get exposed container port
 const containerPort = parser.getContainerPort(); 
```

### CODE REVIEW

File: __tests__/bitbucket_service_test.spec.ts [hunk 1]
```diff
@@ -38,4 +39,23 @@ describe('Bitbucket Service Tests', () => {
...
+      SecretType.NO_AUTH,
+      null
+    );
+
+    const gs = new BitbucketService(gr);
+    gs.getDockerfileContent()
+      .then((content: string) => {
+        const parser = new DockerFileParser(content);
+        const port = parser.getContainerPort();
+        expect(port).toEqual(5000);
```
Comment (DhritiShikhar) [inline __tests__/bitbucket_service_test.spec.ts]: @akashshinde shouldn't there be negative test scenario as well e.g.
- to check if a proper error is returned when Dockerfile is not present

Comment (akashshinde) [reply __tests__/bitbucket_service_test.spec.ts]: Added negative tests https://github.com/redhat-developer/git-service/pull/5/commits/c8bef3d1d7961e40d37501b1e9ad914ec6f8c75a

File: __tests__/github_service_test.ts [hunk 2]
```diff
@@ -115,4 +116,49 @@ describe("Github Tests" , () => {
...
+      .then((content: string) => {
+        const parser = new DockerFileParser(content);
+        const port = parser.getContainerPort();
+        expect(port).toEqual(5000);
+        done();
+      })
+      .catch((err: Error) => done(err))
+  });
+
+  it('should detect Dockerfile', (done: any) => {
```
Comment (DhritiShikhar) [inline __tests__/github_service_test.ts]: I see there is a test `should not return exposed container port` for BitBucket. A similar test for Github is missing.

File: __tests__/github_service_test.ts [hunk 3]
```diff
@@ -115,4 +116,49 @@ describe("Github Tests" , () => {
       })
   });
 
+  it('should return exposed container port', (done: any) => {
```
Comment (DhritiShikhar) [inline __tests__/github_service_test.ts]: I see a test `should not return exposed container port` for Bitbucket. A similar test for Github is missing.

File: src/client_test.ts [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
-const parser = new DockerFileParser(contents);
-parser.parse();
+const gr = new GitSource(
+  "https://bitbucket.org/akshinde/testgitsource",
+  SecretType.NO_AUTH,
+  null
+);
+
+const gs = new BitbucketService(gr);
+gs.getDockerfileContent()
```
Comment (DhritiShikhar) [inline src/client_test.ts]: Shouldn't there be a check for `isDockerfilePresent()` as well?

**Pull request merged**